src/main/scala/firrtl/NewEmitter.scala

```diff
@@ -0,0 +1,344 @@
+package firrtl
+
+import firrtl.ir._
+import firrtl.Utils._
+import firrtl.Mappers._
+import firrtl.PrimOps._
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+object CompoundExprEmitter {
+  def genVerilogReferenceName(expr: Expression): Option[String] = {
+    expr match {
+      case Reference(name, tpe) => Some(name)
```

Comment:
Why are you matching on Reference instead of WRef? Same for SubField/WSubField, WSubIndex/SubIndex?